src/compiler/diagnosticMessages.json

```diff
@@ -3059,6 +3059,10 @@
         "category": "Error",
         "code": 2796
     },
+    "The declaration was marked as deprecated here.": {

```

Comment:
I think it's misleading for the top-line message of a function with some non-deprecated signatures to be "'f' is deprecated". I'd rather see a new message "The signature '(): void' of 'f' is deprecated."

(The message should still be "f is deprecated" for functions where every signature is deprecated.)